### ISSUE TITLE
how does sffs do power loss recovery?

### ISSUE THREAD

**Issue by oska874** [whoozle/sffs#1]
I read the code of sffs,but I didnot find how does sffs do power loss recovery.
If a power loss before I finish writing a file ,how does sffs do when start second time ?How deal with the data of the file ?discard the whole file or ?

**Comment by whoozle** [comment]
It did not do any recovery at all, it just ensures that your data have been written, then switch bit in header. 
So, if you'd never (hopefully) get corrupted data

**Comment by oska874** [comment]
So if there is many file, one file wrote without commit,then only this file discard.

another question, I did not find you erase the flash, and do nothing about garbage collection,did you not implement these ?

thanks a lot.

**Comment by whoozle** [comment]
This file system is intended only for NOR type of flash(typically with SPI
interface, if you need something for NAND, please use ubifs or analogue.
23 июля 2015 г. 11:32 AM пользователь "Ezio Zhang" notifications@github.com
написал:

**Comment by oska874** [comment]
yes. I'm trying to use SFFS on NOR flash in bare system. Well,nor flash file system need garbage collection and erase.
thanks.

**Comment by whoozle** [comment]
Oh really? Well I could add erase handling for you. Take a note that this
is not a file system actually, more like key-value storage. I used it on te
very simple SPI chip which doesn't require erase, or driver took care for
me :)
23 июля 2015 г. 11:47 AM пользователь "Ezio Zhang" notifications@github.com
написал:

**Comment by oska874** [comment]
Another question:did you set block's size very big,as large as flash's erase unit, so driver can do erase for you? Otherwise ,your spi flash is more like a SD card  than a simple flash.

**Comment by whoozle** [comment]
yes, the same size for erase/write blocks
